packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,436 @@
+import ValueState from "@ui5/webcomponents-base/dist/types/ValueState.js";
+import { getFeature } from "@ui5/webcomponents-base/dist/FeaturesRegistry.js";
+import CalendarDate from "@ui5/webcomponents-localization/dist/dates/CalendarDate.js";
+import DateRangePickerTemplate from "./generated/templates/DateRangePickerTemplate.lit.js";
+
+// Styles
+import DateRangePickerCss from "./generated/themes/DateRangePicker.css.js";
+import DatePicker from "./DatePicker.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-daterange-picker",
+	properties: /** @lends sap.ui.webcomponents.main.DateRangePicker.prototype */ {
+		/**
+		 * Determines the symbol which separates the dates.
+		 *
+		 * @type {string}
+		 * @defaultvalue "-"
+		 * @public
+		 */
+		delimiter: {
+			type: String,
+			defaultValue: "-",
+		},
+		/**
+		 * Defines the UNIX timestamp of the first date - seconds since 00:00:00 UTC on Jan 1, 1970.
+		 * @type {number}
+		 * @private
+		*/
+		_firstDateTimestamp: {
+			type: Integer,
+		},
+		/**
+		 * Defines the UNIX timestamp of the second date- seconds since 00:00:00 UTC on Jan 1, 1970.
+		 * @type {number}
+		 * @private
+		*/
+		_lastDateTimestamp: {
+			type: Integer,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.main.DateRangePicker.prototype */ {
+		//
+	},
+	events: /** @lends sap.ui.webcomponents.main.DateRangePicker.prototype */ {
+		//
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ * The DateRangePicker enables the users to enter a localized date range using touch, mouse, keyboard input, or by selecting a date range in the calendar.
+ *
+ * <h3>Usage</h3>
+ * The user can enter a date by:
+ * Using the calendar that opens in a popup or typing it in directly in the input field (not available for mobile devices).
+ * For the <code>ui5-date-range-picker</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/DateRangePicker.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.DateRangePicker
+ * @extends DatePicker
+ * @tagname ui5-daterange-picker
+ * @since 1.0.0-rc.8
+ * @public
+ */
+class DateRangePicker extends DatePicker {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return [...DatePicker.styles, DateRangePickerCss];
+	}
+
+	static get template() {
+		return DateRangePickerTemplate;
+	}
+
+	static async onDefine() {
+		await DatePicker.define();
+	}
+
+	constructor() {
+		super();
+		this.isFirstDatePick = true;
+	}
+
+	_handleInputLiveChange() {
+		const nextValue = this._getInput().getInputValue(),
+			emptyValue = nextValue === "",
+			isValid = emptyValue || (this.isValid(nextValue) && this.isInValidRange(nextValue));
+
+		this.value = nextValue;
+		this.fireEvent("input", { value: nextValue, valid: isValid });
+	}
+
+	onBeforeRendering() {
+		this._calendar.primaryCalendarType = this._primaryCalendarType;
```

Comment:
Is this onBeforeRendering different from the one in the DatePicker?

Reply:
Here the difference is on line 125.